packages/grid/_modules_/grid/hooks/features/sorting/useSorting.ts

```diff
@@ -214,12 +214,16 @@ export const useSorting = (apiRef: ApiRef, rowsProp: RowsProp) => {
   );
 
   const onRowsUpdated = React.useCallback(() => {
-    apiRef.current.applySorting();
-  }, [apiRef]);
+    if (options.sortingMode === FeatureModeConstant.client) {
```

Comment:
Well we can put that check, in `applySorting` directly, it will avoid to repeat it 4 times in the file and also be prone to more issue like this in the future.

Reply:
You know that thought crossed my mind a few times writing this and I kept glazing over it! Changed, thanks for the feedback.